src/test/java/io/jenkins/plugins/analysis/warnings/ReferenceFinderITest.java

```diff
@@ -0,0 +1,740 @@
+     *
+     * @return the created job
+     */
+    private FreeStyleProject createJobWithWorkspaceFile(String name, final String... fileNames) {
+        FreeStyleProject job = createJob(name);
+        copyFilesToWorkspace(job, fileNames);
+        return job;
+    }
+
+    /**
+     * Enables the warnings plugin for the specified job. I.e., it registers a new {@link IssuesRecorder } recorder for
+     * the job.
+     *
+     * @param job
+     *         the job to register the recorder for
+     *
+     * @return the created recorder
+     */
+    @CanIgnoreReturnValue
+    private IssuesRecorder enableWarnings(final FreeStyleProject job) {
+        IssuesRecorder publisher = new IssuesRecorder();
+        publisher.setTools(Collections.singletonList(new ToolConfiguration("**/*issues.txt", new Eclipse())));
+        job.getPublishersList().add(publisher);
+        return publisher;
+    }
+
+    /**
+     * Enables the warnings plugin for the specified job. I.e., it registers a new {@link IssuesRecorder } recorder for
+     * the job.
+     *
+     * @param job
+     *         the job to register the recorder for
+     * @param configuration
+     *         configuration of the recorder
+     *
+     * @return the created recorder
+     */
+    @CanIgnoreReturnValue
+    private IssuesRecorder enableWarnings(final FreeStyleProject job, final Consumer<IssuesRecorder> configuration) {
+        IssuesRecorder publisher = enableWarnings(job);
+        configuration.accept(publisher);
+        return publisher;
+    }
+
+    /**
+     * Schedules a new build for the specified job and returns the created {@link AnalysisResult} after the build has
+     * been finished.
+     *
+     * @param job
+     *         the job to schedule
+     * @param status
+     *         the expected result for the build
+     *
+     * @return the created {@link ResultAction}
+     */
+    @SuppressWarnings({"illegalcatch", "OverlyBroadCatchBlock"})
+    private AnalysisResult scheduleBuildAndAssertStatus(final FreeStyleProject job, final Result status) {
+        try {
+            FreeStyleBuild build = j.assertBuildStatus(status, job.scheduleBuild2(0));
+
+            ResultAction action = build.getAction(ResultAction.class);
+
+            assertThat(action).isNotNull();
+
+            return action.getResult();
+        }
+        catch (Exception e) {
+            throw new AssertionError(e);
+        }
+    }
+
+    @SuppressWarnings({"illegalcatch", "OverlyBroadCatchBlock"})
+    private AnalysisResult scheduleBuildAndAssertStatusWithMultiplePublisher(final FreeStyleProject job,
+            final Result status) {
+        try {
+            FreeStyleBuild build = j.assertBuildStatus(status, job.scheduleBuild2(0));
+
+            List<ResultAction> actions = build.getActions(ResultAction.class);
+            for (ResultAction action : actions) {
+                System.out.println("INFO: Result: " + action.getResult());
```

Comment:
Why are the other actions not important?

Reply:
Kann raus